File: 0709.转换成小写字母/0709-转换成小写字母.c
char * toLowerCase(char * s){
    if(s == NULL)
    {
        return NULL;
    }

    for (int i = 0; i < strlen(s); i++)
    {
        if (s[i] >= 'A' && s[i] <= 'Z')
        {
            //s[i] = (char)(s[i] + 32);//方法 1
            s[i] = (char)(s[i] ^ 32);//方法 2
        } 
    }
    
    return s;
}

File: 0709.转换成小写字母/0709-转换成小写字母.java
class Solution {
    public String toLowerCase(String s) {
        /*
        String res = s.toLowerCase();
        return res;
        */

        if (s == null && "".equals(s))
        {
            return null;
        }
        
        StringBuilder res = new StringBuilder();
        for (char c : s.toCharArray())
        {
            if (c >= 'A' && c <= 'Z')
            {
                //res.append((char)(c + 32));//方法 1
                res.append((char)(c ^ 32));//方法 2
            } 
            else
            {
                res.append(c);
            }
        }
        
        return res.toString();
    }
}